Docs 2.0 New Documentation. Batch 1 of content for review (#672)
graknlabs/docs: 0001-typedb/02-dev/05-read.md

(`$e`). The `get` clause then filters the results to receive only the `email` (`$e`) attributes.

<!-- test-ignore -->
```typeql
match
  $pe ($x, $y) isa permission;
  $x isa person, has full-name $x-n;
  $x-n contains "Kevin";
  $y (accessed-object: $o, valid-action: $act) isa access;
  $act has action-name $act-n;
  $o has path $o-fp;
get $x-n, $act-n, $o-fp;
```

The example above uses a `match` clause to do the following:

1. Find `permission` relation (`$pe`) of `$x` and `$y` variables.
2. Finds `$x` as `person` entities that have `full-name` attribute with value of `$x-n`. 
3. `$x-n` should contain string `Kevin`.
4. Finds `access` relation of `$o` as accessed-object and `$act` as valid-action.
5. `$act` should have `action-name` attribute with value of `$act-n`.
6. `$o` should have `path` attribute with value of `$o-fp`.

The `get` clause then filters the result from the `match` clause to get only the `person`’s `full-name` (`$x-n`), 
`action`’s `action-name` (`$act-n`) and `path` of an `object` (`$o-fp`). Every returned result should contain all 
three variables.

### Modifiers

#### Limit the results

> Use the `limit` keyword followed by the maximum number of results to limit the number of results returned.

Reviewer: ```suggestion
Use the `limit` keyword followed by a positive integer to limit the number of results returned.
```

<!-- test-ignore -->
```typeql
match $p isa person; 
get $p;
limit 1;
```

> This query returns only one single (and random) instance of type `person`. Consider using `limit` with 

Reviewer: ```suggestion
This query returns only one single (and random) instance of type `person`. We strongly recommend using `limit` with 
```

Author: I think I would prefer the original variant here. I think it's closer to the intended message here. Do you think it looks weird or something?

[sorting aggregation](#sort-the-answers) to receive less random and more predictable results.

#### Sort the Answers

Use the `sort` keyword followed by a variable, to sort the answers by the variable. A second argument is optional 

> and determines the sorting order: `asc` (ascending) or `desc` (descending). By default it’s ascending.

Reviewer: ```suggestion
and determines the sorting order: `asc` (ascending, the default) or `desc` (descending).
```

<!-- test-ignore -->
```typeql
match $p isa person, has full-name $n; 
get $n; 
sort $n asc;
```

This query returns all `full-name` attributes of all `person` entities, sorted by their `full-name` in ascending order.

To sort by multiple variables use the same syntax and add additional variables and optional sorting order arguments 
with a comma separator. 

For example:

<!-- test-ignore -->
```typeql
match $p isa person, has full-name $n, has email $e; 
get $n, $e; 
sort $n asc, $e desc;
```

The example above will return all `full-name` and `email` attributes of all `person` entities, sorted by their 
`full-name` in ascending order first and then by `email` in descending order.

#### Offset the Answers

Use the `offset` keyword followed by the number to offset the answers by. This is commonly used with the `limit` 

> keyword to return a desired range of the answers. Don’t forget to [sort](#sort-the-answers) the results to guarantee 

Reviewer: ```suggestion
keyword to return a desired range of the answers. Don’t forget to [sort](#sort-the-answers) the results to ensure 
```

> they will stay in the same order while you move through them.

Reviewer: ```suggestion
predictable and deterministic results.
```

```

If you query the example above, only attribute `$s` will be returned with its value. But all the duplicated values of 
`$s` will be omitted. Because what we are querying is all attribute instances. And when two different entities have 
the same attribute (with the same value) it's still only one attribute.

So it’s just like to query only unique values. For example, if there are three files with the same value of `size-kb` 
attribute = 10, we will get only one of this value.

if we want to prevent that we should return not only attributes, but also the entities that own the attributes in the 
`get` clause. For example:

<!-- test-ignore -->
```typeql 
match $f isa file, has size-kb $s; get $f, $s;
```

This will return pairs of `$f` and `$s`, including all duplicated values of `$s` (every `file` instance with its 
attribute of `size-kb`. Including all of the same values owned by different files. For example, all three values of 10 
will be returned this way).
</div>

### Count

Use the count keyword to get the number of the specified matched variable.

<!-- test-ignore -->
```typeql 
match
  $o isa object, has path $fp;
get $o, $fp; count;
```

<div class="note">
[Note]

> The `count` function is applied to every result returned. If more than one variable mentioned in get, than count 

Reviewer: ```suggestion
The `count` function is applied to every result returned. If more than one variable mentioned in get, then `count` 
```

> will show the number of unique results. This is also the case, when no `get` clause added, which actually means that 

Reviewer: ```suggestion
will show the number of unique results. This is also the case when no `get` clause is added, which actually means that 
```
